api/internal/handler/route/route.go

```diff
@@ -570,30 +557,26 @@ func (h *Handler) Exist(c droplet.Context) (interface{}, error) {
 	input := c.Input().(*ExistCheckInput)
 	name := input.Name
 	exclude := input.Exclude
-	routeStore := store.GetStore(store.HubKeyRoute)
 
-	ret, err := routeStore.List(c.Context(), store.ListInput{
-		Predicate:  nil,
+	ret, err := h.routeStore.List(c.Context(), store.ListInput{
+		Predicate: func(obj interface{}) bool {
+			r := obj.(*entity.Route)
+			if r.Name == name && r.ID != exclude {
+				return true
+			}
+
+			return false
+		},
 		PageSize:   0,
 		PageNumber: 0,
 	})
-
 	if err != nil {
 		return nil, err
 	}
 
-	sort := store.NewSort(nil)
```

Comment:
maybe we could remove these functions now ? I’m not sure if it’s used anywhere else

Reply:
I had checked this. It was still used in upstream.go now. Maybe next time we can delete them.